src/OrleansRuntime/MembershipService/GrainBasedMembershipTable.cs

```diff
@@ -1,28 +1,113 @@
+using System;
+using System.Diagnostics;
 using System.Threading.Tasks;
 using Microsoft.Extensions.DependencyInjection;
 using Orleans.Concurrency;
 using Orleans.MultiCluster;
-using Orleans.Runtime.Configuration;
 using Orleans.Serialization;
 using Microsoft.Extensions.Logging;
 
 namespace Orleans.Runtime.MembershipService
 {
-    //empty marker class inject into DI to see if user configure Silo to use MembershiptableGrain
-    internal class UseGrainBasedMembershipFlag
+    internal class GrainBasedMembershipTable : IMembershipTable
     {
+        private readonly IServiceProvider serviceProvider;
+        private readonly Logger logger;
+        private IMembershipTableGrain grain;
+
+        public GrainBasedMembershipTable(IServiceProvider serviceProvider, LoggerWrapper<MembershipTableFactory> logger)
```

Comment:
Why are we using LoggerWrapper instead of ILogger?

Reply:
I just moved the code that was using `Logger` from `MembershipTableFactory` without unnecessary changes to keep the diff to a minimum (as it's already hard to follow when moving methods from file to file). I can certainly do the switch to `ILogger` if you already diffed and reviewed the code, if there's value of doing it now for this occurrence before the big switch to `ILogger`. Just let me know what to do.

Reply:
sure no strong objection on this one .